components/scream/src/mct_coupling/scream_scorpio_interface.F90

```diff
@@ -450,13 +481,35 @@ subroutine eam_pio_createfile(File,fname)
     retval = pio_createfile(pio_subsystem,File,pio_iotype,fname,mode) 
 
   end subroutine eam_pio_createfile
+!=====================================================================!
+  subroutine eam_pio_openfile(File,fname)
+
+    type(file_desc_t), intent(inout) :: File             ! Pio file Handle
+    character(len=*),  intent(in)    :: fname            ! Pio file name
+    !--
+    integer                          :: retval           ! PIO error return value
+    integer                          :: mode             ! Mode for how to handle the new file
+
+    mode = 0 ! TODO: make sure this is correct. 
+    retval = pio_openfile(pio_subsystem,File,pio_iotype,fname,mode) ! TODO, add error check to use retval, also for createfile below
```

Comment:
I think it would be good to add such check to this pr. It doesn't have to go through all possible error codes, just check whether retval signals an error (of any kind) or not.